src/rust/engine/src/externs/nailgun.rs

```diff
@@ -44,6 +44,9 @@ impl PyNailgunClient {
   fn execute(&self, command: String, args: Vec<String>, env: &PyDict, py: Python) -> PyResult<i32> {
     use nailgun::NailgunClientError;
 
+    // NB: We assume that env var names and values are Python strs strictly convertible to UTF-8
```

Comment:
It was substantially easier to push this responsibility to the Python side rather than verify this on the Rust side, which AFAICT would require a lot of mucking about with the implementation of `FromPyObject::extract`, and adding overhead even in the normal case where we are justified in expecting valid strings.  Instead we say "anything you expect to convert to a Rust String must be a Python str that is strictly encodable to UTF-8." LMK if you think we should do the hoop-jumping in Rust.